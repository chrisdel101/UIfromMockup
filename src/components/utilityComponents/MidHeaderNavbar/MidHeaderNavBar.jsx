import React from 'react'
import './index.css'

function renderNavMarkup(props) {
  return (
    <React.Fragment>
      <div className="mid-navbar">
        {props.links.map((link, i) => {
          return (
            <nav key={i} className="mid-navbar-link">
              <div className="midnav-status-line" />
              <a href={`/${link}`}>{link}</a>
            </nav>
          )
        })}
      </div>
    </React.Fragment>
  )
}
function MidNavBar(props) {
  return <React.Fragment>{renderNavMarkup(props)}</React.Fragment>
}

export default MidNavBar
