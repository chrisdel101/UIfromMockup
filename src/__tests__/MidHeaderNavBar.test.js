import React from 'react'
import MidHeaderNavBar from '../components/utilityComponents/MidHeaderNavBar/MidHeaderNavBar.jsx'
import renderer from 'react-test-renderer'

test('create and match snapshot', () => {
  const links = ['Intro', 'Work', 'About', 'Contacts']
  const component = renderer.create(<MidHeaderNavBar links={links} />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
