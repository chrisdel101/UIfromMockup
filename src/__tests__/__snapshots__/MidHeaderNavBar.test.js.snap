// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create and match snapshot 1`] = `
<div
  className="mid-navbar"
>
  <div
    className="midnav-link-container "
  >
    <nav
      className="mid-navbar-link"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    >
      <a
        href="/Intro"
      >
        <span
          className="mid-nav-number"
        >
          0
          1
        </span>
         
        Intro
      </a>
    </nav>
  </div>
  <div
    className="midnav-link-container "
  >
    <nav
      className="mid-navbar-link"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    >
      <a
        href="/Work"
      >
        <span
          className="mid-nav-number"
        >
          0
          2
        </span>
         
        Work
      </a>
    </nav>
  </div>
  <div
    className="midnav-link-container "
  >
    <nav
      className="mid-navbar-link"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    >
      <a
        href="/About"
      >
        <span
          className="mid-nav-number"
        >
          0
          3
        </span>
         
        About
      </a>
    </nav>
  </div>
  <div
    className="midnav-link-container "
  >
    <nav
      className="mid-navbar-link"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    >
      <a
        href="/Contacts"
      >
        <span
          className="mid-nav-number"
        >
          0
          4
        </span>
         
        Contacts
      </a>
    </nav>
  </div>
</div>
`;
